src/Bedrock.Framework/Transports/Memory/MemoryTransport.cs

```diff
@@ -16,14 +16,14 @@ public partial class MemoryTransport : IConnectionListenerFactory, IConnectionFa
         public ValueTask<IConnectionListener> BindAsync(EndPoint endpoint, CancellationToken cancellationToken = default)
         {
             endpoint ??= MemoryEndPoint.Default;
+            MemoryConnectionListener listener;
 
-            if (_listeners.TryGetValue(endpoint, out _))
+            if (_listeners.TryGetValue(endpoint, out _) ||
+                !_listeners.TryAdd(endpoint, listener = new MemoryConnectionListener() { EndPoint = endpoint }))
```

Comment:
This looks fine, but I wonder if we should be using GetOrAdd or AddOrUpdate to detect this in a single call.